enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+message, etc.).
+
+### Goals
+
+- New Prometheus metric with labels containing aforementioned properties of
+  the recommendation that will indicate if the recommendation is currently active.
+- New Prometheus alert based on this metric, defined using a template that fills
+  the information from labels into the description of the alert.
+- The user should be able to find out when was a certain recommendation active
+  by querying the Prometheus metric (and plot this in Grafana, for example).
+- Configuration option allowing the user to disable/enable this functionality.
+  Since this feature consumes resources on both sides (cluster and Smart Proxy),
+  it should only be enabled on clusters where the user actually wants to use it. (It should be disabled by default, unless the user explicitly enables it.)
+- The enabling should be done using the same approach as the rest of the
+  Insights Operator configuration (i.e., currently using the `support` secret).
+
+### Non-Goals
+
+- Insights Operator fetching additional data, from the Smart Proxy or any other source.
+- Modifications or additions to the user notification systems (the goal is to
+  utilize the existing notification systems based on Prometheus alerts - most
+  notably Alertmanager).
+
+## Proposal
+
+### User Stories
+
+As a user, I want to be immediately notified about a recommendation that has
+become active on my cluster without having to periodically check the Insights
+tab. I want to receive notifications through my preferred channel (e.g., Slack,
+email) configured via Alertmanager when a recommendation becomes active, and
+another one when it becomes inactive again.
+
+### Why should this be implemented in Insights Operator?
+
+The Insights Operator is already fetching the relevant data from the Smart Proxy,
+and managing Prometheus metrics based on this data. Therefore, it makes sense
+for additional information about recommendations to be implemented alongside
+what already exists.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints [optional]
+
+Please note that IO refers to the Insights Operator in this context.
+
+1. IO is enabled and periodically uploads archives to the Ingress service,
+   which are then processed by the CCX data pipeline. (standard behavior)
+2. After a successful archive upload, IO waits for the report to become
+   available, fetches it from the Smart Proxy and processes it. (This is an
+   extension of the current behavior, where we only look at the number of
+   active recommendations per severity level.) (needs to be implemented)
+3. In-cluster Prometheus metrics indicating the status of recommendations
+   being active are set accordingly.
+    - If a recommendation was previously registered as active and is present
+      in the latest Insights report, IO performs no additional action for this
+      recommendation (do not repeat alerts for previously spotted recommendations)
```

Comment:
The way it is written, it describes the end-user experience rather than how the Insights Operator should behave. These parenthesized notes were not written by me (they were added by a colleague when we discussed the behaviour), and are not necessary anymore, but an earlier design idea assumed that the alerts would be short-lived, and we wouldn't want to "revive" them, hence the note. I can remove this one if you would prefer that.